python/tvm/contrib/sdaccel.py

```diff
@@ -41,15 +35,29 @@ def compile_vhls(code, kernel):
     if platform is None:
         raise RuntimeError("No Xlinx device specified.")
 
-    # build xo
-    args = [xocc, "-c", "-t", target, "--platform", platform, "-o", tmp_xo, "-k", kernel] + \
-           advanced_params + [tmp_cpp]
-    returncode = subprocess.call(args)
-    if returncode != 0:
-        raise RuntimeError("Compile error")
+    tmp_xo_files = []
+    for code, kernel in zip(codes, kernels):
```

Comment:
You used two arrays to store `code` and `funcname` respectively in `codegen_vhls.cc` but zip them here again. It seems unnecessary to me. Could you use `tuple` to bind the `code` with its corresponding `funcname` instead?

Reply:
I'm not a bit sure how to pass an Array of tuple to a PackedFunc.  What I came up with was creating an Array of Array like as follows.  Is there an better way?
```diff
diff --git a/src/codegen/codegen_vhls.cc b/src/codegen/codegen_vhls.cc
index 4b291d6..5b402d8 100644
--- a/src/codegen/codegen_vhls.cc
+++ b/src/codegen/codegen_vhls.cc
@@ -80,7 +80,7 @@ runtime::Module BuildSDAccel(Array<LoweredFunc> funcs) {
   std::string whole_code = cg.Finish();
 
   // Generate source code for compilation.
-  Array<Expr> codes, funcnames;
+  Array<Array<Expr>> params;
   for (LoweredFunc f : funcs) {
     CodeGenVivadoHLS cg;
     cg.Init(output_ssa);
@@ -89,13 +89,15 @@ runtime::Module BuildSDAccel(Array<LoweredFunc> funcs) {
     if (const auto* f = runtime::Registry::Get("tvm_callback_vhls_postproc")) {
       code = (*f)(code).operator std::string();
     }
-    codes.push_back(code);
-    funcnames.push_back(f->name);
+    Array<Expr> param;
+    param.push_back(code);
+    param.push_back(f->name);
+    params.push_back(param);
   }
 
   std::string xclbin;
   if (const auto* f = Registry::Get("tvm_callback_sdaccel_compile")) {
-    xclbin = (*f)(codes, funcnames).operator std::string();
+    xclbin = (*f)(params).operator std::string();
   } else {
     LOG(FATAL) << "Cannot compile Vivado HLS code.";
   }
```

Reply:
I'm not quite sure either, but I felt comfortable with this alternative.